packages/builder/src/pages/builder/app/[application]/design/[screenId]/[componentId]/_components/Component/ComponentSettingsSection.svelte

```diff
@@ -44,29 +49,35 @@
             },
           ]
         : []),
-      ...(customSections || []),
+      ...(JSON.parse(JSON.stringify(customSections)) || []),
```

Comment:
Bit of a stinker here, this was copying component definitions by pointer, and the component identities themselves were getting the `collapsed` property applied to them, which caused weirdness if navigating between two components with the same definition.

Reply:
Err is there no way to avoid this? Doesn't seem right to me at all.